### ISSUE TITLE
[package] pybind11/2.6.1: Build modules inclusion causes target conflict with Conan 1.33.0

### ISSUE THREAD

**Issue by jargonzombies** [conan-io/conan-center-index#4322]
### Package and Environment Details
  * Package Name/Version: **pybind11/2.6.1**
  * Conan version: **conan 1.33.0**

### Conan profile
```
[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=9
compiler.libcxx=libstdc++11
build_type=Release
[options]
[build_requires]
[env]
```

### Steps to reproduce
Use the `cmake_find_package` generator in a recipe with the pybind package and the newest Conan version.

### Logs
CMake Error:
```
CMake Error at [...]/pybind11/2.6.1/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/lib/cmake/pybind11/pybind11Common.cmake:48 (add_library):
  add_library cannot create imported target "pybind11::pybind11" because
  another target with the same name already exists.
```

<details><summary>CMake cause traceback</summary>
Generated Findpybind11.cmake:

```cmake
set(pybind11_BUILD_MODULES_PATHS "[...]/pybind11/2.6.1/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/lib/cmake/pybind11/FindPythonLibsNew.cmake"
			"[...]/pybind11/2.6.1/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/lib/cmake/pybind11/pybind11Install.cmake")
[...]
if(NOT ${CMAKE_VERSION} VERSION_LESS "3.0")
    # Target approach
    if(NOT TARGET pybind11::pybind11)
        add_library(pybind11::pybind11 INTERFACE IMPORTED)
        [...]
    endif()
endif()

foreach(_BUILD_MODULE_PATH ${pybind11_BUILD_MODULES_PATHS})
    include(${_BUILD_MODULE_PATH})
endforeach()
```

In included pybind11Install.cmake

```cmake
[...]
if(TARGET pybind11::python_link_helper)
  # This has already been setup elsewhere, such as with a previous call or
  # add_subdirectory
  return()
endif()
[...]
include("${CMAKE_CURRENT_LIST_DIR}/pybind11Targets.cmake")
```

In included pybind11Targets.cmake:
```cmake
[...]
# CMake 3.10 has an include_guard command, but we can't use that yet
if(TARGET pybind11::lto)
  return()
endif()
[...]
add_library(pybind11::pybind11 IMPORTED INTERFACE ${optional_global})
```
</details>

In short, although there are two include guards, they reference targets that the generated `Findpybind11.cmake` doesn't create, and the `pybind11::pybind11` target gets created twice as a result.

Comparing conan 1.32.1 and 1.33.0, the build module inclusion has been reordered to after the generation of targets.

## Possible fixes

I don't know how this should be fixed. It's necessary that the pybind build modules are included so that all pybind targets and macros are available and the target conflict needs to be avoided. Arguably, the `pybind11::pybind11` target shouldn't be generated, or the build module inclusion needs to precede the generation of targets by the find module, although this is a conan change instead of a package fix.

**Comment by jargonzombies** [comment]
#4407 is a duplicate, possibly fixed by #4445

**Comment by blackliner** [comment]
Why not just rename the conan global target to something like `DONT_USE_pybind11::DONT_USE_pybind11` and leave the generation of the "real" targets to the modules?

**Comment by danimtb** [comment]
Fixed by #4445
